models/classes/Lists/Business/Service/ValueCollectionService.php

```diff
@@ -91,4 +97,19 @@ public function persist(ValueCollection $valueCollection): bool
 
         return false;
     }
+
+    private function setUserDataLanguage(ValueCollectionSearchRequest $searchRequest): void
+    {
+        $user = $this->getServiceLocator()->get(SessionService::class)->getCurrentUser();
+        /** @var UserLanguageServiceInterface $userLanguageService */
+
+        $userLanguageService = $this->getServiceLocator()
+            ->get(UserLanguageServiceInterface::SERVICE_ID);
+
+        $searchRequest->setDataLanguage(
+            $userLanguageService->getDataLanguage($user)
+        );
```

Comment:
```suggestion
        if ($userLanguageService->isDataLanguageEnabled()) {
            $searchRequest->setDataLanguage(
                $userLanguageService->getDataLanguage($user)
            );
        }
```

As I saw, depending on the configuration `OPTION_LOCK_DATA_LANGUAGE`, we should not allow dataLanguage filter, correct?